src/main/java/it/gov/pagopa/payhub/activities/activity/sendnotification/NotificationStatusActivityImpl.java

```diff
@@ -11,14 +14,29 @@
 @Lazy
 public class NotificationStatusActivityImpl implements NotificationStatusActivity {
     private final SendService sendService;
+    private final DebtPositionService debtPositionService;
+    private final InstallmentService installmentService;
 
-    public NotificationStatusActivityImpl(SendService sendService) {
+    public NotificationStatusActivityImpl(SendService sendService,
+        DebtPositionService debtPositionService,
+        InstallmentService installmentService){
         this.sendService = sendService;
+      this.debtPositionService = debtPositionService;
+      this.installmentService = installmentService;
     }
 
     @Override
     public SendNotificationDTO getSendNotificationStatus(String sendNotificationId) {
         log.info("Starting notificationStatus for sendNotificationId {}", sendNotificationId);
-        return sendService.notificationStatus(sendNotificationId);
+        SendNotificationDTO sendNotificationDTO = sendService.notificationStatus(sendNotificationId);
+        if(sendNotificationDTO!=null && sendNotificationDTO.getIun()!=null) {
+            sendNotificationDTO.getPayments().forEach(p -> {
+                DebtPositionDTO debtPositionDTO = debtPositionService.getDebtPosition(p.getDebtPositionId());
+                debtPositionDTO.getPaymentOptions().forEach(po -> po.getInstallments().stream()
+                    .filter(installment -> installment.getIun() != null)
+                    .forEach(installment -> installmentService.updateIun(installment.getInstallmentId(), sendNotificationDTO.getIun())));
+            });
```

Comment:
This way you are requesting to debt-position to resolve and return the PII, can we avoid?
you could also expose an update API directly on InstallmentEntityExtendedController to accept debtPositionId and nav (or directly a list of nav) and perform the update on a single invocation